### PR TITLE
fix: broken pagination

### LINKED ISSUE #2360: Bug: Pagination is not working properly.
### Describe the bug

Pagination should reset to 1 upon changing the language in Explore page.

### Steps to reproduce

1. Go to **Explore Page**
2. Select any language and scroll down to the pagination
3. Change the page number from **1** to anything in the range, lets say may be **2**
4. Scroll to the top of the page change the language and you will see that the list of repos for the newly selected language are fetched for page **1** but the page number in the pagination is not **1** it is **2** because we selected it before changing the language.
5. Now change the page number to **1** and the list of repos will not change obviously because it is already fetched for page **1**
6. But if you change the page number to **2** again, you will find the list of repos are fetched for page **2**.

> **NOTE**: I used page 2 for this example but this issue is same for any page number other than 1. This issue is also found in other pages where the pagination component is being used.

sample

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by sourabpramanik**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [X] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [X] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #2360 

## Mobile & Desktop Screenshots/Recordings

**Before Fix:** Changing the language shows the list of repos for page number 1 but pagination indicator didn't change to page number 1 and also the page numbers were not calculated correctly
1. Indicating wrong page number
pagination bug #1
2. Unable to use back arrow
pagination bug #1

**After Fix:** Changing the language shows the list of repos for page number 1 and resets the the indicator to 1 and calculates the page numbers in between the arrows
pagination fix #1

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by sourabpramanik** [comment]
@a0m0rajab @nickytonline  can you please review the fix?

File: components/molecules/Pagination/pagination.tsx [hunk 1]
```diff
@@ -121,7 +116,7 @@ const Pagination = ({
         )}
         {goToPage && (
           <div className="hidden md:block">
-            <PaginationGotoPage totalPage={totalPage} page={page} setPage={handleSelected} name={""} />
+            <PaginationGotoPage totalPage={totalPage} page={currentPage} setPage={handleSelected} name={""} />
```
Comment (OgDev-01) [inline components/molecules/Pagination/pagination.tsx:119]: Definitely not for the PR, but the name prop makes no sense here as this should be static. we can name the input directly in https://github.com/sourabpramanik/open-sauced-app/blob/a3c9d2ad460fcefe7c4607ccf96773dcf6e73514/components/molecules/PaginationGotoPage/pagination-goto-page.tsx#L33
and remove the name prop. I'll open a ticket for this tiny refactor 👍

**Review by OgDev-01 — DISMISSED** [review]
Works as espected locally 🍕

File: components/molecules/Pagination/pagination.tsx (hunk 1 above)
Comment (sourabpramanik) [reply components/molecules/Pagination/pagination.tsx:119]: Sure will remove that

**Comment by nickytonline** [comment]
Thanks for the PR @sourabpramanik!. Taking a quick glance at this, there is no need to change `page` to `currentPage`. Let's focus on the fix.

**Comment by sourabpramanik** [comment]
> Thanks for the PR @sourabpramanik!. Taking a quick glance at this, there is no need to change `page` to `currentPage`. Let's focus on the fix.

Will revert that. No problem!!

File: components/molecules/Pagination/pagination.tsx (hunk 1 above)
Comment (sourabpramanik) [reply components/molecules/Pagination/pagination.tsx]: @OgDev-01 did you raised the ticket for this refractor?

**Review by nickytonline — APPROVED** [review]
Thanks for the fix @sourabpramanik! :shipit:

**Pull request merged**